### PR TITLE
[#851] Tab bar layout for the home screen

### LINKED ISSUE #851: Tab bar layout for the Home Screen
The Home screen is a single view - navigation view. According to the new design, this has to be a tab view

https://www.figma.com/file/vYQD6uZtkPgtmgJ3zfauq0/Zashi-UI-082823-Final?node-id=489%3A7388&mode=dev

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #851 

- concept of tabs implemented, it's the main navigation hub
- home reducer simplified because it no longer plays a role of a navigation hub
- navigation bar titles implemented
- hamburger menu for the settings screen
- balances screen tweaked
- wallet events on home screen instead of a separate screen
- tests fixed + new tests added
- feature flags removed from the debug menu, no longer used
- semitransparent nav bar implemented

Simulator Screenshot - iPhone 15 Plus - 2023-10-10 at 15 14 59
Simulator Screenshot - iPhone 15 Plus - 2023-10-10 at 15 15 03
Simulator Screenshot - iPhone 15 Plus - 2023-10-10 at 15 15 08
Simulator Screenshot - iPhone 15 Plus - 2023-10-10 at 15 14 56

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [x] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: modules/Sources/Features/AddressDetails/AddressDetailsView.swift [hunk 1]
```diff
@@ -36,11 +36,11 @@ public struct AddressDetailsView: View {
...
-            .zashiBack()
-            .zashiTitle {
-                Text(L10n.AddressDetails.receiveTitle.uppercased())
-                    .font(.custom(FontFamily.Archivo.bold.name, size: 14))
-            }
+//            .zashiBack()
+//            .zashiTitle {
+//                Text(L10n.AddressDetails.receiveTitle.uppercased())
+//                    .font(.custom(FontFamily.Archivo.bold.name, size: 14))
+//            }
```
Comment (HonzaR) [inline modules/Sources/Features/AddressDetails/AddressDetailsView.swift:43]: Should we rather remove it than comment out?

File: secantTests/WalletConfigProviderTests/WalletConfigProviderTests.swift [hunk 2]
```diff
@@ -137,7 +137,7 @@ class WalletConfigProviderTests: XCTestCase {
         store.receive(.updateStateAfterConfigUpdate(flags)) { state in
             state.walletConfig = flags
             state.onboardingState.walletConfig = flags
-            state.homeState.walletConfig = flags
+//            state.homeState.walletConfig = flags
```
Comment (HonzaR) [inline secantTests/WalletConfigProviderTests/WalletConfigProviderTests.swift:140]: Same comment as above.

**Review by HonzaR — APPROVED** [review]
Approved with two minor comments inline.

File: modules/Sources/Features/AddressDetails/AddressDetailsView.swift (hunk 1 above)
Comment (LukasKorba) [reply modules/Sources/Features/AddressDetails/AddressDetailsView.swift:43]: Removed

File: secantTests/WalletConfigProviderTests/WalletConfigProviderTests.swift (hunk 2 above)
Comment (LukasKorba) [reply secantTests/WalletConfigProviderTests/WalletConfigProviderTests.swift:140]: fixed

**Pull request merged**